app/models/query_record.rb

```diff
@@ -15,12 +19,28 @@ def self.cleanup
                   (@@last_cleanup < 5.minutes.ago) ||
                   ::Rails.env.test?
 
-    connection.delete(%(
-      DELETE FROM #{table_name}
-      WHERE
-        access_count = 0 AND updated_at < DATE_SUB(NOW(), INTERVAL 6 HOUR) OR
-        access_count > 0 AND updated_at < DATE_SUB(NOW(), INTERVAL 1 DAY)
-    ))
+    delete_manager = arel_delete_cleanup(table_name)
+    # Nimmo note: Reviewers can examine the generated SQL like so:
+    # puts(delete_manager.to_sql)
+    connection.delete(delete_manager.to_sql)
+
     @@last_cleanup = Time.zone.now
   end
+
+  # Nimmo Note: This seems like a good place to keep it in Arel.
+  # It enables passing the table name as a variable; can't imagine in AR.
+  # Original SQL:
+  # DELETE FROM #{table_name}
+  # WHERE
+  #   access_count = 0 AND updated_at < DATE_SUB(NOW(), INTERVAL 6 HOUR) OR
+  #   access_count > 0 AND updated_at < DATE_SUB(NOW(), INTERVAL 1 DAY)
+  private_class_method def self.arel_delete_cleanup(table_name)
+    table = Arel::Table.new(table_name)
+    Arel::DeleteManager.new.
+      from(table).
+      where(table[:access_count].eq(0).
+        and(table[:updated_at].lt(Time.zone.now - 6.hours)).
+        or(table[:access_count].gt(0).
+        and(table[:updated_at].lt(Time.zone.now - 1.day))))
+  end
```

Comment:
(wow the parentheses are really hard to keep straight in Arel!!)

Reply:
I like the idea of giving it a day and simplifying the conditional. 👍

(Incidentally, and beside the point, check this tricky syntax highlighting. I know you're not a fan on principle, but it keeps me sane...)
![Screen Shot 2022-05-07 at 5 23 41 PM](https://user-images.githubusercontent.com/1948095/167276517-87228c4f-af9b-439e-b019-877ab8dc54d5.png)